### ISSUE TITLE
Event listener should return an appropriate response when pipelinerun fails

### ISSUE THREAD

**Issue by hokadiri** [tektoncd/triggers#1183]
# Expected Behavior
I get an appropriate response that the pipelinerun failed to create (e.g an http 500)
# Actual Behavior
Silently fails with a HTTP 202 code

# Steps to Reproduce the Problem
Introduce a bug in your trigger template. E.g using upper case in the pipelinerun generated name.

```
apiVersion: triggers.tekton.dev/v1beta1
kind: TriggerTemplate
metadata:
  name: pipeline-template
spec:
  params:
  - name: gitrevision
    description: The git revision
    default: main
  - name: gitrepositoryurl
    description: The git repository url
  - name: message
    description: The message to print
    default: This is the default message
  - name: contenttype
    description: The Content-Type of the event
  resourcetemplates:
  - apiVersion: tekton.dev/v1beta1
    kind: PipelineRun
    metadata:
      generateName: Simple-pipeline-run-
    spec:
      pipelineRef:
        name: simple-pipeline
      params:
      - name: message
        value: $(tt.params.message)
      - name: contenttype
        value: $(tt.params.contenttype)
      resources:
      - name: git-source
        resourceSpec:
          type: git
          params:
          - name: revision
            value: $(tt.params.gitrevision)
          - name: url
            value: $(tt.params.gitrepositoryurl)
```
Event listener silently fails (returns a `202`) even though it failed to create the pipelinerun. 
```
time="2021-08-10T14:16:05Z" level=info msg="trigger API responsemap[status_code:202 Accepted]"
```

```
{
  "level": "error",
  "ts": "2021-08-10T14:16:05.124Z",
  "logger": "eventlistener",
  "caller": "sink/sink.go:278",
  "msg": "couldn't create resource with group version kind \"tekton.dev/v1beta1, Resource=pipelineruns\": PipelineRun.tekton.dev \"Simple-pipeline-run-75cpl\" is invalid: [metadata.generateName: Invalid value: \"Simple-pipeline-run-\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), metadata.name: Invalid value: \"Simple-pipeline-run-75cpl\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]",
  "knative.dev/controller": "eventlistener",
  "eventlistener": "some-event-listener",
  "namespace": "android",
  "eventlistenerUID": "df681322-0b4d-4e0f-8015-9e24bec443db",
  "/triggers-eventid": "f74b7d21-e9e9-401b-9524-2f28f0591179",
  "/trigger": "pipeline-template",
  "stacktrace": "github.com/tektoncd/triggers/pkg/sink.Sink.processTrigger\n\tgithub.com/tektoncd/triggers/pkg/sink/sink.go:278\ngithub.com/tektoncd/triggers/pkg/sink.Sink.HandleEvent.func4\n\tgithub.com/tektoncd/triggers/pkg/sink/sink.go:164"
}
```

- Kubernetes version:

  **Output of `kubectl version`:**

```
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.4-dirty", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"dirty", BuildDate:"2021-03-15T10:03:32Z", GoVersion:"go1.16.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.8-gke.900", GitCommit:"28ab8501be88ea42e897ca8514d7cd0b436253d9", GitTreeState:"clean", BuildDate:"2021-06-30T09:23:36Z", GoVersion:"go1.15.13b5", Compiler:"gc", Platform:"linux/amd64"}
```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

```
Client version: 0.17.1
Pipeline version: v0.24.1
Triggers version: v0.14.0
Dashboard version: v0.17.0
```

**Comment by dibyom** [comment]
Thanks for the detailed info @hokadiri  The HTTP 202 behavior is intentional but I think there are other ways to accomplish your goal.

Some background - A single incoming event can trigger multiple triggers. Previously we used to wait for all triggers to finish processing and then respond with a 201. A few issues with this: 
1. Even if we returned a 201 it does not mean each trigger was processed successfully - some of them could have failed while others succeeded. 

2. The EventListener has to wait for all triggers to finish processing. This can take a while e.g. if there are a number of interceptors, or if there is a network issues. General best practices for authoring webhook receivers indicate that we should respond as quickly as possible else the webhook may be retried e.g. see GitHub's guidance [here]
(https://docs.github.com/en/rest/guides/best-practices-for-integrators#favor-asynchronous-work-over-synchronous).

So, since Triggers are processed asynchronously, it is not always possible to return the result of Trigger processing as part of the HTTP response. What we do instead is return the eventID which can be used to [find any resources that were created](https://github.com/tektoncd/triggers/tree/main/examples#invoking-the-triggers-locally) . The logs also contain the eventID in order to help debug when something goes wrong. Finally, we are working on emitting Kubernetes events when processing succeeds or fails which should provide another way to find out when Trigger processing [failed](https://github.com/tektoncd/triggers/issues/649).
